src/Yoti.Auth/Activity.cs

```diff
@@ -40,21 +40,54 @@ public ActivityDetails HandleSuccessfulResponse(AsymmetricCipherKeyPair keyPair,
             {
                 ReceiptDO receipt = parsedResponse.Receipt;
 
-                AttrpubapiV1.AttributeList attributes = CryptoEngine.DecryptCurrentUserReceipt(
-                    parsedResponse.Receipt.wrapped_receipt_key,
-                    parsedResponse.Receipt.other_party_profile_content,
-                    keyPair);
+                ParseOtherPartyProfileContent(keyPair, parsedResponse);
+                var applicationProfileAttributes = ParseApplicationProfileContent(keyPair, parsedResponse);
 
-                _yotiUserProfile.Id = parsedResponse.Receipt.remember_me_id;
-                _yotiProfile.Id = parsedResponse.Receipt.remember_me_id;
-                string receiptId = parsedResponse.Receipt.receipt_id;
+                ApplicationProfile applicationProfile = new ApplicationProfile(applicationProfileAttributes);
+                DateTime? timestamp = null;
 
-                AddAttributesToProfile(attributes);
+                if (DateTime.TryParseExact(parsedResponse.Receipt.timestamp, "yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime parsedDate))
```

Comment:
1. Can _timestamp_ be null?
2. What happens if parsing fails?  Does _TryParseExact_ throw an Exception?

Reply:
1. Will add null check
2. [Looks like](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tryparseexact?view=netframework-4.7.2) it can just throw an `ArgumentException`